tests/e2e/tests/suite.go

```diff
@@ -91,6 +91,14 @@ type Suite struct {
 }
 
 func (s *Suite) NewTest(t *testing.T) *Test {
+	if s == nil {
```

Comment:
How comes that we now have these sanity checks added in the context of this PR?
Don't get me wrong - I'm all for making the software more robust but it's not obvious to me where this is coming from when "only" adding a train example to an existing test-case.

Reply:
Removed the sanity checks from this PR.